### DISCUSSION TITLE
Unresolvable dependency error with Caching config

### DISCUSSION THREAD

**Discussion by LTKort** [spatie/laravel-ray#283] [Q&A] [answered]
**Describe the bug**
Suddenly when I try to use artisan optimize or artisan config:cache, I get the following error:
```php
In Container.php line 1106:                                                                                                                
  Unresolvable dependency resolving [Parameter #0 [ <required> array $settings ]] in class Spatie\Ray\Settings\Settings  
```

**Versions**
Ray version (you can see this in "About Ray"):

You can use `composer show` to get the version numbers of:
- spatie/ray package version: v1.36
- spatie/laravel-ray package version (if applicable): v1.32

PHP version: 8.0 & 8.1
Laravel version: v9.4

**To Reproduce**
Run commands 'php artisan optimize' or 'php artisan config:cache'

**Expected behavior**
No Error

**Desktop (please complete the following information):**
 - OS: [macOS & on server]
 - Version Latest
 
 **Additional context**

I'm using the latest version of Laravel, and got the error on multiple projects, servers & devices.

Ray is not installed as a dev dependency, and the error occurs with & without the config being published.
I checked the results of the SettingFactory and the config seems to be loaded.

I tried it using a fresh installation and everything works

How is this possible?

**Comment by LTKort** [comment, marked as answer]
Ahh, I think I got the answer. There was a package trying to access the config in the register() of the ServiceProvider. Which caused the console to give the error linked to Ray

**Reply by bilogic** [reply]
> There was a package trying to access the config in the register()

how did you arrive at this conclusion?

**Reply by LTKort** [reply]
@bilogic I don't remember exactly. But i think by commenting out service providers and installing packages one by one until the error occurred.

**Reply by bilogic** [reply]
ok thanks
